background/style-manager.js

```diff
@@ -345,6 +349,56 @@ const styleMan = (() => {
     });
   }
 
+  function handlePublishingUSW(port) {
+    if (port.name !== 'link-style-usw') {
+      return;
+    }
+    port.onMessage.addListener(async incData => {
+      const {data: style, reason} = incData;
+      if (!style.id) {
+        return;
+      }
+      switch (reason) {
+        case 'revoke':
+          await tokenMan.revokeToken('userstylesworld', style.id);
+          style._usw = {};
+          handleSave(await saveStyle(style), 'success-revoke', true);
+          break;
+
+        case 'publish':
+          if (!style._usw || !style._usw.token) {
+            // Ensures just the style does have the _isUswLinked property as `true`.
+            for (const {style: someStyle} of dataMap.values()) {
+              if (someStyle._id === style._id) {
+                someStyle._isUswLinked = true;
+                someStyle.sourceCode = style.sourceCode;
+                const {metadata} = await API.worker.parseUsercssMeta(style.sourceCode);
+                someStyle.metadata = metadata;
+              } else {
+                delete someStyle._isUswLinked;
+                delete someStyle.sourceCode;
+                delete someStyle.metadata;
+              }
+              handleSave(await saveStyle(someStyle), null, null, false);
+            }
+            style._usw = {
+              token: await tokenMan.getToken('userstylesworld', true, style),
+            };
```

Comment:
@Gusted
1. I don't see where this `metadata` property is used.
2. Saving `metadata` into the database is wrong.
3. The code resaves all styles regardless of whether they have those temporary properties or not.

I want to fix these problems in my current PR.

Reply:
Here's its new place: https://github.com/openstyles/stylus/pull/1285/files#diff-a1faa9074772a4ebc2143e7ea6a934984c561d3e339e5d4132da82b09e56a60dR398

Reply:
> * I don't see where this `metadata` property is used.
> 
> * Saving `metadata` into the database is wrong.

They are used on USw to pre-fill certain information extracted from the style itself(We don't want to import usercss-meta ourself if it can be done on Stylus). It's one of the things to make it more easily to add new styles from existing styles from Stylus. And as I hadn't really any option to pass down just that kind of information to the install-usw-hook.js I've figured to just save it into the db temporarily. 

> The code resaves all styles regardless of whether they have those temporary properties or not.

This was a bit of a design problem I didn't really thought about. As it just worked and didn't touch it after that.

Reply:
So it looks like only one style can be linked? I think this is wrong so I'll try passing the data directly and thus lift the restriction.

Reply:
Ah, I get it: a single style has the mark for the current communication, but it doesn't mean that only one style can be linked to USW.

Reply:
Ah yes sorry for not responding, got pretty must wasted from covid-19 vaccinate.

But that was pretty much the goal of that code, to ensure only 1 is marked for communication(in case stylus was aborted while such property's still existed on the style).